src/formats/apple-notes.ts

```diff
@@ -0,0 +1,307 @@
+			defaultPath: dataPath,
+			properties: ['openDirectory'],
+			//see https://developer.apple.com/videos/play/wwdc2019/701/
+			message: 'Select the "group.com.apple.notes" folder to allow Obsidian to read Apple Notes data.'
+		});
+		
+		if (!names?.includes(dataPath)) {
+			new Notice('Data import failed. Ensure you have selected the correct Apple Notes data folder.');
+			return null;
+		}
+				
+		const originalDB = path.join(dataPath, NOTE_DB);
+		const clonedDB = path.join(os.tmpdir(), NOTE_DB);
+
+		fs.copyFileSync(originalDB, clonedDB);
+		fs.copyFileSync(originalDB + '-shm', clonedDB + '-shm');
+		fs.copyFileSync(originalDB + '-wal', clonedDB + '-wal');
+
+		return new SQLiteTagSpawned(clonedDB, { readonly: true, persistent: true });
+	}
+
+	async import(ctx: ImportContext): Promise<void> {
+		this.ctx = ctx;
+		this.rootFolder = await this.getOutputFolder() as TFolder;
+		this.attachmentPath = this.getAttachmentPath();
+		
+		if (!this.rootFolder) {
+			new Notice('Please select a location to export to.');
+			return;
+		}
+		
+		this.database = await this.getNotesDatabase() as SQLiteTagSpawned;
+		if (!this.database) return;
+				
+		const noteAccounts = await this.database.all`
+			SELECT z_pk FROM ziccloudsyncingobject WHERE z_ent = 13 /* account entity */
+		`;
+		const noteFolders = await this.database.all`
+			SELECT z_pk FROM ziccloudsyncingobject WHERE z_ent = 14 /* folder entity */
+		`;
+		
+		for (let a of noteAccounts) await this.resolveAccount(a.Z_PK);
+		for (let f of noteFolders) await this.resolveFolder(f.Z_PK);
+		
+		const notes = await this.database.all`
+			SELECT
+				z_pk, zfolder, ztitle1 FROM ziccloudsyncingobject 
+			WHERE
+				z_ent = 11 /* Note entity */
+				AND ztitle1 IS NOT NULL
+				AND zfolder != ${this.trashFolder}
+		`;
+		this.noteCount = notes.length;
+		
+		for (let n of notes) {
+			try { 
+				await this.resolveNote(n.Z_PK); 
+			}
+			catch (e) { 
+				this.ctx.reportFailed(n.ZTITLE1, e?.message); 
+			}
+		}
+	}
+	
+	async resolveAccount(id: number): Promise<void> {
+		if (!this.multiAccount && Object.keys(this.resolvedAccounts).length) {
+			this.multiAccount = true;
+		}
+		
+		const account = await this.database.get`
+			SELECT zname, zidentifier FROM ziccloudsyncingobject
+			WHERE z_ent = 13 AND z_pk = ${id}
+		`;
+			
+		this.resolvedAccounts[id] = {
+			name: account.ZNAME,
+			uuid: account.ZIDENTIFIER
+		};
+	}
+	
+	async resolveFolder(id: number): Promise<void> {
+		if (id in this.resolvedFiles) return;
+		
+		const folder = await this.database.get`
+			SELECT ztitle2, zparent, zidentifier, zfoldertype, zowner
+			FROM ziccloudsyncingobject
+			WHERE z_ent = 14 AND z_pk = ${id}
+		`;
+		let prefix;
+		
+		if (folder.ZFOLDERTYPE == ANFolderType.Smart) {
+			return;	
+		}
+		else if (!this.importTrashed && folder.ZFOLDERTYPE == ANFolderType.Trash) {
+			this.trashFolder = id;
+			return;	
+		}
+		else if (folder.ZPARENT !== null) {
+			this.resolveFolder(folder.ZPARENT);
```

Comment:
Missing `await` here. This caused the next line to fail in some cases when testing.